### ISSUE TITLE
USB  keys being automounted despite udiskie "don't automount" setting

### ISSUE THREAD

**Issue by kdv666** [coldfix/udiskie#209]
Hello.
Udiskie was working fine for me, but since I upgraded to Ubuntu-Mate 20.04, I can't seem to cancel automatic mounting of USB keys when they are plugged in.  A "df" command shows them mounted in the usual "media/<user>/<label> place, and the udiskie tray icon shows them mounted.

This is my .config/udiskie/config.yml file.  I'd appreciate suggestions.

program_options:
  tray:               true
  menu:               flat
  automount:          false
  notify:             false
  file_manager:       /usr/bin/caja

device_config:
- device_file:        /dev/loop*
  ignore:             true
- id_uuid:            9ff9aac8-8422-46a5-ac39-4b24b6dc42ab
  ignore:             true

**Comment by coldfix** [comment]
Hi,

are you sure that it's being automounted *by udiskie*, and not by some other part of the system? What is the terminal output of udiskie while this is going on?

Best, Thomas

**Comment by kdv666** [comment]
Hello.  Thanks for your reply.  I'm not sure where it's being mounted.  I turned off automounting in the dconf-editor org.gnome.desktop.media-handling, and in the caja file browsing.  I watched the output of udevadm and can't see the culprit. So I assumed that udisks2 might be doing it, and hence udiskie would block that.

No terminal output when I plug the drive in, and it mounts.  If I tell udiskie to unmount (from the tray), it announces in the terminal that it is doing so.

Regards, Kenneth

**Comment by coldfix** [comment]
> No terminal output when I plug the drive in, and it mounts. If I tell udiskie to unmount (from the tray), it announces in the terminal that it is doing so.

Sounds like udiskie isn't responsible for mounting it.

Some other process may be using udisks2 or another method.

If this is going over udisks2, you may be able to find out which process is doing it as follows:

- Run `dbus-monitor --system interface=org.freedesktop.UDisks2.Filesystem`. If the mount is performed using udisks, you will see output similar to:

```
method call time=1598825053.648043 sender=:1.46 -> destination=:1.47 serial=53 path=/org/freedesktop/UDisks2/block_devices/sdb1; interface=org.freedesktop.UDisks2.Filesystem; member=Mount
   array [
      dict entry(
         string "options"
         variant             string ""
      )
   ]
```

- Look for the `sender=:1.46`, and find its PID:

```
$ dbus-send --system --print-reply --dest=org.freedesktop.DBus /org/freedesktop/DBus org.freedesktop.DBus.GetConnectionUnixProcessID 'string::1.46'
method return time=1598825757.475302 sender=org.freedesktop.DBus -> destination=:1.2533 serial=3 reply_serial=2
   uint32 605
```

- Finally, check what process this is, e.g. `ps -p 605`

**Comment by kdv666** [comment]
Thank you for your detailed reply.  I  take my hat off to your expertise!

Ok, it's fixed now.  I did as you suggested, and the end result was "gvfs-udisks2-vo".  That's something that never came up in my google searches for the automounting behaviour.

I went back into the dconf-editor, and did a general search for "automount'.  I had already disabled "org/gnome/desktop/media-handling/automount.  It turns out that Ubuntu-mate had put in another one, "org/mate/desktop/media-handling/automount".  Once I turned that one off, udiskie started behaving normally.  So you put me on the right track.

My error was in thinking that anything that used udisks2  was intercepted by udiskie.  My bad.  I greatly appreciate your taking the time to help and to put me on the right track.  All's well in the world, at least until the next time I bump against the ceiling of my ignorance! :-)
